src/main/java/de/azapps/kafkabackup/common/TopicConfiguration.java

```diff
@@ -0,0 +1,20 @@
+package de.azapps.kafkabackup.common;
+
+import java.util.Map;
+import java.util.TreeMap;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@Getter
+public class TopicConfiguration {
+
+  private final String topicName;
+  private final int partitionsNumber;
+  private Map<String, String> configuration;
+
+  public void setConfiguration(Map<String, String> configuration) {
+    this.configuration = new TreeMap<>(configuration);
```

Comment:
Is the TreeMap what ensures that the config will be serialised in key-order?

Reply:
yes, sorted by keys